src/basicstruct.py

```diff
@@ -62,6 +77,15 @@ def __ne__(self, other):
     def __hash__(self):
         return hash(self._to_tuple())
 
+    def __iter__(self):
+        """Yield pairs of (attrubute_name, value).
+
+        This allows using `dict(my_struct)`.
+
+        """
+        for key in self.__slots__:
```

Comment:
`dict(iterable)` expects pairs of `(key, value)`. `iter(self._to_tuple)` will yield only values.